plugins/dap/dap-ui.nix

```diff
@@ -0,0 +1,209 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.dap.extensions.dap-ui;
+  helpers = import ../helpers.nix {inherit lib;};
+
+  mkSizeOption =
+    helpers.mkNullOrOption
+    (
+      with types;
+        either int (numbers.between 0.0 1.0)
+    );
+
+  mkMappingOption =
+    helpers.defaultNullOpts.mkNullable
+    (
+      with types;
+        attrsOf (either str (listOf str))
+    );
+
+  layoutOption = types.submodule {
+    options = {
+      elements = mkOption {
+        default = [];
+        description = "Elements to display in this layout.";
+        type = types.listOf types.attrs;
+      };
+
+      size = mkOption {
+        default = 10;
+        description = "Size of the layout in lines/columns.";
+        type = types.int;
+      };
+
+      position = mkOption {
+        default = "left";
+        description = "Which side of editor to open layout on.";
+        type = types.enum ["left" "right" "top" "bottom"];
+      };
+
+      id = helpers.mkNullOrOption types.str "Element ID.";
+    };
+  };
+in {
+  options.plugins.dap.extensions.dap-ui =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "dap-ui";
+
+      package = helpers.mkPackageOption "dap-ui" pkgs.vimPlugins.nvim-dap-ui;
+
+      controls = helpers.mkCompositeOption "Options for dap-ui controls." {
+        enabled = helpers.defaultNullOpts.mkBool true "Enable controls";
+
+        element =
+          helpers.defaultNullOpts.mkEnumFirstDefault ["repl" "scopes" "stacks" "watches" "breakpoints" "console"]
+          "Element to show the controls on.";
+
+        icons = {
+          disconnect = helpers.defaultNullOpts.mkStr "" "";
+          pause = helpers.defaultNullOpts.mkStr "" "";
+          play = helpers.defaultNullOpts.mkStr "" "";
+          run_last = helpers.defaultNullOpts.mkStr "" "";
+          step_into = helpers.defaultNullOpts.mkStr "" "";
+          step_over = helpers.defaultNullOpts.mkStr "" "";
+          step_out = helpers.defaultNullOpts.mkStr "" "";
+          step_back = helpers.defaultNullOpts.mkStr "" "";
+          terminate = helpers.defaultNullOpts.mkStr "" "";
+        };
+      };
+
+      elementMappings =
+        helpers.mkNullOrOption
+        (
+          with types;
+            attrsOf (
+              attrsOf (either str (listOf str))
```

Comment:
Its maybe more a submodule than an attrsOf ? 
The keys only seems to be able to be `"expand"|"open"|"remove"|"edit"|"repl"|"toggle"` according to [the source](https://github.com/rcarriga/nvim-dap-ui/blob/c020f660b02772f9f3d11f599fefad3268628a9e/lua/dapui/config/init.lua#L9).